api/client_test.go

```diff
@@ -113,3 +118,70 @@ func TestClientURL(t *testing.T) {
 		}
 	}
 }
+
+func TestDoGetFallback(t *testing.T) {
+	v := url.Values{"a": []string{"1", "2"}}
+
+	type testResponse struct {
+		Values string
+		Method string
+	}
+
+	// Start a local HTTP server
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		req.ParseForm()
+		r := &testResponse{
+			Values: req.Form.Encode(),
+			Method: req.Method,
+		}
+
+		body, _ := json.Marshal(r)
+
+		if req.Method == http.MethodPost {
+			if req.URL.Path == "/blockPost" {
+				http.Error(w, string(body), http.StatusMethodNotAllowed)
+				return
+			}
+		}
+
+		w.Write(body)
+	}))
+	// Close the server when test finishes
+	defer server.Close()
+
+	u, err := url.Parse(server.URL)
+	testutil.Ok(t, err)
+	client := &httpClient{client: *(server.Client())}
+
+	// Do a post, and ensure that the post succeeds
+	_, b, err := DoGetFallback(client, context.TODO(), u, v)
+	if err != nil {
+		t.Fatalf("Error doing local request: %v", err)
+	}
+	resp := &testResponse{}
+	if err := json.Unmarshal(b, resp); err != nil {
+		t.Fatal(err)
```

Comment:
why not use testutil everywhere?

Reply:
I've swapped the plain `t.Fatal(err)` to testutil, the rest provide more helpful error messages about the failure

Reply:
you can use testutil.Assert with a message.

I think only .Ok doesn't support a message, but it shows the failing line so you can always go there and check.